### DISCUSSION TITLE
Storage Prometheus data outside pod, PV errors

### DISCUSSION THREAD

**Discussion by mgorzkow** [prometheus-operator/kube-prometheus#1989] [Q&A] [answered]
Hi, how can i storage data outside the prometheus-k8s-0 pod? 

I tried to do PV by this instrunction (Step 5) https://computingforgeeks.com/setup-prometheus-and-grafana-on-kubernetes/
but i get errors like this:

+ rm -rf manifests
+ mkdir -p manifests/setup
+ jsonnet -J vendor -m manifests prometheus-pvc.jsonnet
+ xargs '-I{}' sh -c 'sleep .2; cat {} | gojsontoyaml > {}.yaml' -- '{}'
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error
--: 1: gojsontoyaml: Exec format error

or this:

+ mkdir -p manifests/setup
+ jsonnet -J vendor -m manifests prometheus-pvc.jsonnet
./build.sh: line 18: jsonnet: command not found
+ xargs '-I{}' sh -c 'sleep .2; cat {} | gojsontoyaml > {}.yaml' -- '{}'

**Comment by mgorzkow** [comment]
Now im getting this error
--: /usr/local/bin/gojsontoyaml: cannot execute binary file

Can anyone help please?

**Comment by mgorzkow** [comment, marked as answer]
finally i made it work

**Reply by phiconsultores** [reply]
What was the solution?

**Reply by DangNguyen146** [reply]
I fix this by way 

```
VER=$(curl -s https://api.github.com/repos/brancz/gojsontoyaml/releases/latest | grep tag_name | cut -d '"' -f 4 | sed 's/v//')
OS=$(uname -s | tr '[:upper:]' '[:lower:]')

wget https://github.com/brancz/gojsontoyaml/releases/download/v${VER}/gojsontoyaml_${VER}_${OS}_amd64.tar.gz
tar xvf gojsontoyaml_${VER}_${OS}_amd64.tar.gz
sudo mv gojsontoyaml /usr/local/bin
```
It have to solution two dowload pakage
```
wget https://github.com/brancz/gojsontoyaml/releases/download/v${VER}/gojsontoyaml_${VER}_darwin_amd64.tar.gz
wget https://github.com/brancz/gojsontoyaml/releases/download/v${VER}/gojsontoyaml_${VER}_linux_amd64.tar.gz
```
You must choose architecture compatible `dpkg --print-architecture`
